modules/021-cni-cilium/images/safe-agent-updater/main.go

```diff
@@ -0,0 +1,247 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	ciliumNS             = "d8-cni-cilium"
+	generationAnnotation = "safe-agent-updater-daemonset-generation"
+	scanInterval         = 3 * time.Second
+	scanIterations       = 20
+)
+
+func main() {
+	config, _ := rest.InClusterConfig()
+	kubeClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("[SafeAgentUpdater] Failed to init kubeClient. Error: %v", err)
+	}
+	nodeName := os.Getenv("NODE_NAME")
+	if len(nodeName) == 0 {
+		log.Fatalf("[SafeAgentUpdater] Failed to get env NODE_NAME.")
+	}
+	currentPodName, isCurrentGenEqDesiredGen, err := checkGeneration(kubeClient, nodeName)
```

Comment:
```suggestion
	currentAgentPodName, isCurrentAgentPodGenerationDesired, err := checkAgentPodGeneration(kubeClient, nodeName)
```

curren
i don't like condition in varname. isAgentPodGenerationDesired.